contentScript.js

```diff
@@ -3,103 +3,121 @@ let counter = 0;
+    )[0];
 
-  let following = false;
-  console.log("pid", profileId);
-  const sessionId = document.cookie.match(/JSESSIONID="([^"]*)/)[1];
-
-  document
-    .querySelectorAll(".pvs-profile-actions .display-flex.t-normal.flex-1")
-    .forEach((e) => {
-      if (e.textContent == "Following") following = true;
-    });
-
-
-  if (following && !document.getElementById("linkedin-unfollow")) {
-    document
-      .getElementsByClassName("pvs-profile-actions")[0]
-      .insertAdjacentHTML(
-        "beforeend",
-        '<button id="linkedin-unfollow" class="artdeco-button artdeco-button--2 artdeco-button--secondary ember-view pvs-profile-actions__action"><span class="artdeco-button__text">UF</span></button>'
-      );
+    let following = false;
+    console.log("pid", profileId);
+    const sessionId = document.cookie.match(/JSESSIONID="([^"]*)/)[1];
 
     document
-      .getElementById("linkedin-unfollow")
-      .addEventListener("click", () => {
-        sendRequest(profileId, sessionId);
-        document.getElementById("linkedin-unfollow").disabled = true;
-      });
-  } else if (!following && !document.getElementById("linkedin-follow")) {
+        .querySelectorAll(".pvs-profile-actions .display-flex.t-normal.flex-1")
+        .forEach((e) => {
+            if (
+                e.textContent == "Following" &&
+                connRequest?.innerText !== "Accept"
+            )
+                following = true;
+        });
     document
-      .getElementsByClassName("pvs-profile-actions")[0]
-      .insertAdjacentHTML(
-        "beforeend",
-        '<button id="linkedin-follow" class="artdeco-button artdeco-button--2 artdeco-button--secondary ember-view pvs-profile-actions__action"><span class="artdeco-button__text">F</span></button>'
-      );
-
-    document.getElementById("linkedin-follow").addEventListener("click", () => {
-      sendRequest(profileId, sessionId, "true");
-      document.getElementById("linkedin-follow").disabled = true;
+        .getElementsByClassName("pvs-profile-actions")[0]
+        .insertAdjacentHTML(
+            "beforeend",
+            '<button id="custom-linkedin-follow" class="artdeco-button artdeco-button--2 artdeco-button--secondary ember-view pvs-profile-actions__action"><span class="artdeco-button__text">UF</span></button>'
+        );
+    const customLinkedinFollowButton = document.getElementById(
+        "custom-linkedin-follow"
+    );
+    const sendRequest = async () => {
+        customLinkedinFollowButton.disabled = true;
+        try {
+            const response = await fetch(
+                `https://${window.location.host}/voyager/api/feed/dash/followingStates/urn:li:fsd_followingState:urn:li:fsd_profile:${profileId}`,
+                {
+                    headers: {
+                        "csrf-token": sessionId,
+                    },
+                    body: `{"patch":{"$set":{"following":${!following}}}}`,

```

Comment:
with this logic, If I'm not following someone, and I accept their request, I will follow them?